### PR TITLE
posix: eventfd: Fix unsetting internal flags in ioctl

### LINKED ISSUE #71399: posix: eventfd: Setting flags with ioctl F_SETFL operation deletes internal flags
**Describe the bug**
Using the `F_SETFL` operation of `ioctl` on an `eventfd` to set flags results in internal flag state being deleted; This includes the internal `EFD_IN_USE_INTERNAL` flag, which is used to determine whether the `eventfd` has been opened. Thus, any further operation on the file descriptor fails.

This is a regression that has been introduced in commit e6eb0a705bc8 ("posix: eventfd: revise locking, signaling, and allocation").

**To Reproduce**
```
int fd = eventfd(0, 0);

ioctl(fd, F_SETFL, O_NONBLOCK);

eventfd_write(fixture->fd, 19); /* fails */
```

**Expected behavior**
After using `F_SETFL` on an open `eventfd`, regardless of whether it was successful, the target `eventfd` should still be a valid file descriptor, and subsequent operations should work as expected.

**Impact**
Annoyance. `eventfd` cannot be reconfigured after opening without reopening it.

**Logs and console output**
n/a

**Environment (please complete the following information):**
 - OS: Debian
 - Toolchain: Zephyr, West
 - Commit SHA: cb99303dd44c

**Additional context**
Fix in PR #71344

### PR DESCRIPTION
**Pull request by celinakalus**
Commit e6eb0a705bc8 ("posix: eventfd: revise locking, signaling, and allocation") introduced a regression where the internal flags of an event file descriptor would be erased when calling the F_SETFL ioctl operation.

This includes the flag EFD_IN_USE_INTERNAL which determines whether this file descriptor has been opened, thus effectively closing the eventfd whenever one tries to change a flag.

Fixes #71399

### CODE REVIEW

File: lib/posix/options/eventfd.c [hunk 1]
```diff
@@ -247,7 +247,9 @@ static int eventfd_ioctl_op(void *obj, unsigned int request, va_list args)
 			errno = EINVAL;
 			ret = -1;
 		} else {
-			efd->flags = flags;
+			int prev_flags = efd->flags & ~EFD_FLAGS_SET_INTERNAL;
+
+			efd->flags = flags | prev_flags;
```
Comment (jukkar) [inline lib/posix/options/eventfd.c:252]: We could avoid extra variable by this:
```
efd->flags = (efd->flags & ~EFD_FLAGS_SET_INTERNAL ) | flags;
```
The line is still readable and not even that long.

Comment (celinakalus) [reply lib/posix/options/eventfd.c:252]: Thanks for your feedback. Personally I prefer the separate variable, but if that is an issue, of course I'll change it.

**Review by cfriedt — COMMENTED** [review]
The changes look fine (I'm not terribly opinionated about the additional variable).

However, since it has been identified as a regression, it would be much appreciated if you could add a second commit to this PR to test for the correct / expected functionality in the eventfd testsuite.

With that, at least CI failures will prevent this type of regression in the future.

Thanks for the fix 😊

**Comment by ycsin** [comment]
..and probably a bug report so that it can be backported?

**Comment by celinakalus** [comment]
I added a test and a bug report so it can be backported.

**Review by cfriedt — APPROVED** [review]
LGTM 👍

**Review by ycsin — APPROVED** [review]

**Pull request merged**